src/com/goide/inspections/GoStringCannotBeNilInspection.java

```diff
@@ -0,0 +1,90 @@
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class GoStringCannotBeNilInspection extends GoInspectionBase{
+  public static final String QUICK_FIX_NAME = "Change to default value";
+  public static final String DEFAULT_STRING = "\"\"";
+  public static final String PROBLEM_DESCRIPTION = "String cannot be nil";
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor(){
+
+      @Override
+      public void visitVarSpec(@NotNull GoVarSpec o) {
+        super.visitVarSpec(o);
+
+        if(o.getVarDefinitionList() == null) return;
+        o.getVarDefinitionList().forEach(var -> check(var, var != null ? var.getValue() : null));
+      }
+
+      @Override
+      public void visitAssignmentStatement(@NotNull GoAssignmentStatement o) {
+        super.visitAssignmentStatement(o);
+
+        if(o.getLeftHandExprList() == null) return;
+        List<GoExpression> rightSide = o.getExpressionList();
+        List<GoExpression> leftSide = o.getLeftHandExprList().getExpressionList();
+        if(leftSide == null || rightSide == null) return;
+
+        IntStream.range(0 , Math.min(leftSide.size(), rightSide.size()))
+          .forEach(i -> check(leftSide.get(i), rightSide.get(i)));
+      }
+
+      @Override
+      public void visitBinaryExpr(@NotNull GoBinaryExpr o) {
+        super.visitBinaryExpr(o);
+        if(o.getOperator() == null || o.getOperator().getNode() == null) return;
+        IElementType type = o.getOperator().getNode().getElementType();
+        if(type == GoTypes.EQ || type == GoTypes.NOT_EQ){
+          check(o.getLeft(), o.getRight());
+          check(o.getRight(), o.getLeft());
+        }
+      }
+
+      protected void check(GoTypeOwner var, GoExpression value){
+
+        if(var == null || value == null) return;
+        GoType varType = var.getGoType(null);
+        if(!GoTypeUtil.isString(varType)) return;
+        PsiReference valueRef = value.getReference();
+        if(valueRef == null) return;
+        PsiElement valueResolved = valueRef.resolve();
+        if(value.textMatches(GoConstants.NIL) && valueResolved != null && GoPsiImplUtil.builtin(valueResolved)){
+          holder.registerProblem(value, PROBLEM_DESCRIPTION, ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                                 new GoChangeStringToDefaultValueQuickFix());
+        }
+      }
+    };
+  }
+
+  public static class GoChangeStringToDefaultValueQuickFix extends LocalQuickFixBase {
+
+    public GoChangeStringToDefaultValueQuickFix() {
+      super(QUICK_FIX_NAME);
+    }
+
+    @Override
+    public void applyFix(@NotNull Project project, @NotNull ProblemDescriptor descriptor) {
+      PsiElement element = descriptor.getPsiElement();
+      if(element == null) return;
+      element.replace(GoElementFactory.createExpression(project, DEFAULT_STRING));
```

Comment:
`element` can be invalid at this point. Also you don't check what element it is. E.g. in case of `batch` inspection mode that was run on injected text, you'll get a file here as an element, I don't think you want to replace file with `""`